### PR TITLE
New package for Carbon Black EDR logs

### PR DESCRIPTION
**Pull request by adriansr**
## What does this PR do?
Adds a new package for VMware Carbon Black EDR logs ingested via [CB Event Forwarder.](https://github.com/carbonblack/cb-event-forwarder)

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Screenshots

image

Screenshot 2021-08-15 at 13-32-08 VMware Carbon Black EDR - Integrations - Elastic

Default enabled data_streams:

Screenshot 2021-08-15 at 13-33-52 Add integration - VMware Carbon Black EDR - Integrations - Elastic

Extra data_streams:

Screenshot 2021-08-15 at 13-34-38 Add integration - VMware Carbon Black EDR - Integrations - Elastic

## What's missing

- [ ] Logo, can't be used without explicit agreement from VMware.
- [ ] Dashboards.

### CODE REVIEW

File: packages/carbonback_edr/changelog.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# newer versions go on top
+- version: "0.1.0"
+  changes:
+    - description: initial release
+      type: enhancement # can be one of: enhancement, bugfix, breaking-change
+      link: https://github.com/elastic/integrations/pull/NNN
```
Comment (adriansr) [inline packages/carbonback_edr/changelog.yml:6]: Update this

File: packages/carbonback_edr/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+release: beta
+description: Carbon Black EDR Integration
+type: integration
+format_version: 1.0.0
+license: basic
+categories: [security]
+conditions:
+  kibana.version: "^7.14.0"
+# TODO
+#icons:
```
Comment (adriansr) [inline packages/carbonback_edr/manifest.yml:13]: Find an usable icon or remove this

**Review by leehinman — CHANGES_REQUESTED** [review]
looks great.  2 minor things ecs.version bump and related.hash.  one optional.

File: packages/carbonback_edr/_dev/build/build.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+dependencies:
+  ecs:
+    reference: git@1.10
```
[the remark this answers is not in the document]
Comment (adriansr) [reply packages/carbonback_edr/_dev/build/build.yml]: Done

File: packages/carbonback_edr/data_stream/log/_dev/test/pipeline/test-events.json-expected.json [hunk 4]
```diff
@@ -0,0 +1,4496 @@
...
+            "observer": {
+                "name": "CB_SERVER_HOSTNAME",
+                "product": "Carbon Black EDR",
+                "type": "ips",
+                "vendor": "VMWare"
+            },
+            "@timestamp": "2021-06-25T10:32:46.182Z",
+            "carbonblack": {
+                "edr": {
+                    "sha256": "BCA92084BAC4A371631307A8015E0AB5F221CD0593D6D589007F2AF689B0C6FA",
```
[the remark this answers is not in the document]
Comment (adriansr) [reply packages/carbonback_edr/data_stream/log/_dev/test/pipeline/test-events.json-expected.json:13]: Done

File: packages/carbonback_edr/data_stream/log/_dev/test/pipeline/test-events.json-expected.json [hunk 5]
```diff
@@ -0,0 +1,4496 @@
...
+                    "scores": {},
+                    "host": "CB_SERVER_IP",
+                    "watchlists": {},
+                    "link_md5": "https://CB_SERVER/#/binary/B1EADD12A85FD321B05819041C27CC7B",
+                    "group": "SENSORS_GROUP_NAME",
+                    "md5": "B1EADD12A85FD321B05819041C27CC7B",
```
[the remark this answers is not in the document]
Comment (adriansr) [reply packages/carbonback_edr/data_stream/log/_dev/test/pipeline/test-events.json-expected.json:19]: Done

File: packages/carbonback_edr/data_stream/log/manifest.yml [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+        multi: false
+        required: false
+        show_user: true
+      - name: ssl
+        type: yaml
+        title: SSL
+        multi: false
+        required: false
+        show_user: false
+      - name: tags
```
[the remark this answers is not in the document]
Comment (adriansr) [reply packages/carbonback_edr/data_stream/log/manifest.yml]: I didn't know this was possible. Done

[the remark this answers is not in the document]
Comment (adriansr) [reply packages/carbonback_edr/data_stream/log/manifest.yml:30]: Turns out I can't get it to work. If I move those vars to the data_stream's top level either on the main manifest or the data_stream's manifest, there's no error but they don't show up in the UI. Do you have a working example?

[the remark this answers is not in the document]
Comment (leehinman) [reply packages/carbonback_edr/data_stream/log/manifest.yml:30]: For history I was thinking of what we did for zeek, but that is one variable for multiple data streams.  Looks like we don't have a way to do one variable for multiple inputs.

**Review by leehinman — APPROVED** [review]
LGTM

**Comment by CyberTaoFlow** [comment]
no support for carbon black cloud events

**Pull request merged**